modules/dhcp_common/subnet_service.rb

```diff
@@ -2,9 +2,11 @@ module Proxy::DHCP
   class SubnetService
     include Proxy::Log
 
+    SEARCH_MASKS = (0..31).map { |bit| ~(1 << bit) }
```

Comment:
It did make a noticeable difference, even though it's a trivial operation. Every little calculation does seem to matter (I also removed a slightly superfluous `& 0xff` from ipv4_to_i too, it saved a small amount of work.)